esp32-wroom-rp/src/wifi.rs

```diff
@@ -1 +1,71 @@
 pub use crate::spi::Wifi;
+
+use defmt::{write, Format, Formatter};
+
+/// An enumerated type that represents the current WiFi network connection status.
+#[repr(u8)]
+#[derive(Eq, PartialEq, PartialOrd, Debug)]
+pub enum ConnectionStatus {
+    /// No device is connected to hardware
+    NoEsp32 = 255,
+    /// Temporary status while attempting to connect to WiFi network
+    Idle = 0,
+    /// No SSID is available
+    NoActiveSsid,
+    /// WiFi network scan has finished
+    ScanCompleted,
+    /// Device is connected to WiFi network
+    Connected,
+    /// Device failed to connect to WiFi network
+    Failed,
+    /// Device lost connection to WiFi network
+    Lost,
+    /// Device disconnected from WiFi network
+    Disconnected,
+    /// Device is listening for connections in Access Point mode
+    ApListening,
+    /// Device is connected in Access Point mode
+    ApConnected,
+    /// Device failed to make connection in Access Point mode
+    ApFailed,
+    /// Unexpected value returned from device, reset may be required
+    Invalid,
+}
+
+impl From<u8> for ConnectionStatus {
+    fn from(status: u8) -> ConnectionStatus {
+        match status {
+            0   => ConnectionStatus::Idle,
+            1   => ConnectionStatus::NoActiveSsid,
+            2   => ConnectionStatus::ScanCompleted,
+            3   => ConnectionStatus::Connected,
+            4   => ConnectionStatus::Failed,
+            5   => ConnectionStatus::Lost,
+            6   => ConnectionStatus::Disconnected,
+            7   => ConnectionStatus::ApListening,
+            8   => ConnectionStatus::ApConnected,
+            9   => ConnectionStatus::ApFailed,
+            255 => ConnectionStatus::NoEsp32,
+            _   => ConnectionStatus::Invalid,
+        }
+    }
+}
+
+impl Format for ConnectionStatus {
```

Comment:
@dilyn-corner Are all of these meant as errors? I believe the intent is that these are current status and not all errors. It actually makes me wonder if some of them could actually be broken out into errors vs those that are truly about the current wifi connection status.

@calebbourg what do you think about this?

Reply:
I agree that these are probably not all errors. For example `ConnectionStatus::Connected` should probably be considered a successful result. `ConnectionStatus::Disconnected` I think as well and a few others.  

In terms of the `Format` implementation I wonder if it would make sense to just use the language you used in the [documentation of the enum](https://github.com/Jim-Hodapp-Coaching/esp32-wroom-rp/pull/47/files#diff-41a6d08fa17cbc3493bf8aaaabed097a5ce2a25c860a1253c6efc135f1f0b6b4R8)? 

So for example `ApListening`  would map to`"Device is connected in Access Point mode"`

Reply:
My first thought was to simply return what was documented -- only three of these variants actually feel like errors, so I agree that the suggestion is the right choice.

Reply:
Awesome, so close and looking really great!